Narrow symbols

supascooopa opened the pull request:
<!-- See README for documentation, or ask in #zulip-terminal if unclear -->
### What does this PR do, and why?
Adding symbols to narrow buttons on the top left of the side UI. This is to improve UI visually.





### External discussion & connections
<!-- [x] all that apply, specifying topic and adding numbers after # for issues/PRs -->
- [x] Discussed in **#zulip-terminal** in `topic`
- [ ] Fully fixes #
- [x] Partially fixes issue #1420 
- [ ] Builds upon previous unmerged work in PR #
- [ ] Is a follow-up to work in PR #
- [ ] Requires merge of PR #
- [ ] Merge will enable work on #

### How did you test this?
<!-- [x] all that apply -->
- [ ] Manually - Behavioral changes
- [x] Manually - Visual changes
- [x] Adapting existing automated tests
- [ ] Adding automated tests for new behavior (or missing tests)
- [ ] Existing automated tests should already cover this (*only a refactor of tested code*)

### Self-review checklist for each commit
- [x] It is a [minimal coherent idea](https://github.com/zulip/zulip-terminal#structuring-commits---speeding-up-reviews-merging--development)
- [ ] It has a commit summary following the [documented style](https://github.com/zulip/zulip-terminal#structuring-commits---speeding-up-reviews-merging--development) (title & body)
- [ ] It has a commit summary describing the  motivation and reasoning for the change
- [x] It individually passes linting and tests
- [x ] It contains test additions for any new behavior
- [ ] It flows clearly from a previous branch commit, and/or prepares for the next commit

### Visual changes    <!-- DELETE SECTION IF NO VISUAL CHANGE -->
<!-- Zulip tips at https://zulip.readthedocs.io/en/latest/tutorials/screenshot-and-gif-software.html -->
<!-- For video, try asciinema; after uploading, embed using
[![yourtitle](https://asciinema.org/a/<id>.png)](https://asciinema.org/a/<id>)
-->
<!-- NOTE: Attached videos/images will be clearer from smaller terminal windows -->
Before:
<img width="382" alt="image" src="https://github.com/zulip/zulip-terminal/assets/77788985/90a9e8d0-a822-4798-b78f-630400d3a64e">

After: 
<img width="289" alt="image" src="https://github.com/zulip/zulip-terminal/assets/77788985/1a962cef-bb81-4e5f-876f-f8f881bc7de7">

neiljp commented on zulipterminal/ui_tools/buttons.py:
```diff
@@ -118,7 +125,10 @@ def keypress(self, size: urwid_Size, key: str) -> Optional[str]:
     def __init__(self, *, controller: Any, count: int) -> None:
-        button_text = f"All messages     [{primary_key_for_command('ALL_MESSAGES')}]"
+        button_text = (
+            f"{ALL_MESSAGES_MARKER}"
```
These markers can be handled like with other buttons, by passing a `prefix_markup` parameter to `TopButton`.

neiljp commented on zulipterminal/config/symbols.py:
```diff
@@ -9,6 +9,9 @@
 
 INVALID_MARKER = "✗"  # BALLOT X, U+2717 (Dingbats)
 DIRECT_MESSAGE_MARKER = "§"  # SECTION SIGN, U+00A7 (Latin-1 supplement)
+ALL_MESSAGES_MARKER = "≡"
```
Before merging, it would be good to have a similar comment for this symbol, ie. with the unicode name, hex, etc.

The other symbols you added aren't as important since they're currently standard letters/punctuation.

supascooopa replied:
I'll add a comment similar to the other ones.

neiljp reviewed (commented):
@supascooopa Thanks for pushing this into a PR :+1:

I looked at the tests - we have a flakey (unreliable) test, and restarting just one made them complete.

The symbols look good to me, though the star doesn't look as central vertically compared to the others, so not as good perhaps. One aspect of that we can explore is styling the symbols, maybe just bold or something similar. That is what led to me pointing out the prefix_markup, since it's easier to style the prefix separately from the text that way.

Do you plan to add the symbols to the narrows as viewed in the centre title area too?

supascooopa commented:
> @supascooopa Thanks for pushing this into a PR 👍
> 
> I looked at the tests - we have a flakey (unreliable) test, and restarting just one made them complete.
> 
> The symbols look good to me, though the star doesn't look as central vertically compared to the others, so not as good perhaps. One aspect of that we can explore is styling the symbols, maybe just bold or something similar. That is what led to me pointing out the prefix_markup, since it's easier to style the prefix separately from the text that way.
> 
> Do you plan to add the symbols to the narrows as viewed in the centre title area too?

I'm relived to find out that the test was not working properly and not something I did, I was being real careful to have my ducks in a row before pushing to remote this time.

I will give `prefix_markup` a try and see how the star symbol turns out that way.

I was actually planning to add the prefix symbols in the center title area but I wasn't sure to do it in this PR or next one. To clarify, you mean the area right below 'Messages' title right?

neiljp commented:
@supascooopa If you're happy managing the prefix symbols in this PR then that's fine, as separate later commits to what you have already :+1:

supascooopa commented:
@neiljp I've adjusted the prefix markers according to your suggestions. I wasn't sure how to implement the styling, add a brand new one to the styles or use an existing one so I just went with the latter. 
I'll start working on prefix symbols on the narrow titles when we agree on the details on this area.

neiljp commented:
@supascooopa Thanks for the update :+1:

Structurally the code looks simpler using the prefix markup. I'm not entirely sure the bold makes too much difference for me and is sufficient, but with the common style name, we can easily adjust that at a later point.

Similarly the star marker looks 'high', but we can adjust later if finding a suitable alternative.

To support getting this into a mergeable state, before continuing, I'd suggest combining your most recent commits separately into the earlier two, since they are fixes to each respectively. Then once you have two commits, getting the commit titles according to our standards.

---
If you want to keep the substance of the commits from your previous iteration, then depending upon your comfort level with git, you might find it useful in future to add (or move) the fixing (fixup) commits right after each one you wish to change. That makes it clear if a new commit relates to an update to a specific earlier commit and makes it very simple to combine (fixup or squash) those with the earlier commit later. However, within Zulip it's also totally accepted to push to your branch with a completely fixed/updated set of commits, doing a force-push if necessary.

supascooopa commented:
@neiljp Thank you for the suggestion regarding git I will keep it in mind next time.
 I've squashed the commits you've mentioned, changed the commit messages and pushed them. I hope the commit messages are up to the standards :)

neiljp commented:
@supascooopa This looks fine to move ahead with :+1:

The first commit adds multiple symbols, not just the all-messages symbol, but we can deal with that detail later or on merging.

Are you working on the symbols for the central narrow titles now?

supascooopa commented:
@neiljp Oh yes! I think it would be better to check the changes before writing the message next time. 

I want to work on adding the symbols to the central narrow titles. Should I work on this draft or open a different one?

neiljp commented:
@supascooopa I just tidied up the commit messages and pushed them back here - that way we can get this merged and in wider use.

Please review the differences I made in the commit messages. I also slightly adjusted your additions in the first commit, to ensure the existing and new markers were grouped more cleanly.  You can fetch this PR with `tools/fetch-pull-request`, and compare this to what you already have locally with `git diff` and `git range-diff`

Note that you could have developed beyond this branch in a 2nd local branch ("on top of" this one), as long as you were willing to keep the 2nd one consistent with any changes to this - that's where you can rebase locally, and handle any conflicts. That can work well if you don't expect the first branch to change too much, while allowing you to continue working beyond that and being ready to push another branch/PR.

supascooopa commented:
Thank you very much @neiljp. I will keep your advice in mind for next time.
